collect_app/src/androidTest/java/org/odk/collect/android/support/pages/NotificationDrawer.kt

```diff
@@ -152,15 +163,36 @@ class NotificationDrawer {
         }
     }
 
+    private fun close() {
+        val device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+        if (isOpen) {
+            device.pressBack()
+        }
+
+        ensureClosed()
+    }
+
     /**
      * It appears that sometimes the notification drawer does not close automatically when it should
      * such as after clicking on a notification or its action. This could be due to a bug in Android.
      */
-    private fun closeNotificationDrawerIfOpened() {
+    private fun ensureClosed() {
         val device = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
-        val isManageButtonGone = device.wait(Until.gone(By.text("Manage")), TimeInMs.THREE_SECONDS)
-        if (!isManageButtonGone) {
+        if (!device.wait(Until.gone(DRAWER_SEARCH_CONDITION), TimeInMs.ONE_SECOND * 3)) {
             device.pressBack()
+            assertThat(
+                "Couldn't close notification drawer!",
+                device.wait(Until.gone(DRAWER_SEARCH_CONDITION), TimeInMs.ONE_SECOND),
+                equalTo(true)
+            )
         }
+
+        this.isOpen = false
```

Comment:
Why? It doesn't change anything here.

Reply:
There are a few places where we use `isOpen` to determine behaviour (`teardown` for example), so keeping it up to date is important.